webapp/metrics/metrics.py

```diff
@@ -53,6 +53,54 @@ def _calculate_color(thisCountry, maxCountry, maxColor, minColor):
     return int(colorRange * countryFactor + minColor)
 
 
+def _capitalize_os_name(os_name):
+    """Capitalize OS name
+
+    :returns: Capitalized OS name (if part of the list)
+    """
+    capitalized_oses = {
+        "fedora/31": "Fedora 31",
+        "ubuntu-core/18": "Ubuntu Core 16",
+        "linuxmint/18": "Linux Mint 18",
+        "parrot/4.7": "Parrot OS 4.7",
+        "centos/7": "CentOS 7",
+        "ubuntu/16.10": "Ubuntu 16.10",
+        "arch": "Arch Linux",
+        "fedora/28": "Fedora 28",
+        "linuxmint/18.1": "Linux Mint 18.1",
+        "debian": "Debian",
+        "elementary/5.1": "elementary OS 5.1",
+        "linuxmint/18.2": "Linux Mint 18.2",
+        "ubuntu/20.04": "Ubuntu 20.04",
+        "fedora/29": "Fedora 29",
+        "elementary/0.4.1": "elementary OS 0.4.1",
+        "neon/16.04": "KDE Neon 16.04",
+        "linuxmint/19.3": "Linux Mint 19.3",
+        "fedora/30": "Fedora 30",
+        "debian/10": "Debian 10",
+        "ubuntu/14.04": "Ubuntu 14.04",
+        "manjaro": "Manjaro",
+        "ubuntu/17.04": "Ubuntu 17.04",
+        "zorin/15": "Zorin OS 15",
+        "linuxmint/19.2": "Linux Mint 19.2",
+        "elementary/5.0": "elementary OS 5.0",
+        "debian/9": "Debian 9",
+        "zorin/12": "Zorin OS 12",
+        "linuxmint/19": "Linux Mint 19",
+        "linuxmint/18.3": "Linux Mint 18.3",
+        "neon/18.04": "KDE Neon 18.04",
+        "linuxmint/19.1": "Linux Mint 19.1",
+        "ubuntu/17.10": "Ubuntu 17.10",
+        "ubuntu-core/16": "Ubuntu Core 16",
+        "ubuntu/19.10": "Ubuntu 19.10",
+        "ubuntu/18.10": "Ubuntu 18.10",
+        "ubuntu/19.04": "Ubuntu 19.04",
+        "ubuntu/16.04": "Ubuntu 16.04",
+        "ubuntu/18.04": "Ubuntu 18.04",
+    }
+    return capitalized_oses.get(os_name, os_name)
```

Comment:
I think we can probably reduce this hardcoded list down to something like:
```
  "fedora": "Fedora",
  "ubuntu-core": "Ubuntu Core",
  "ubuntu": "Ubuntu",
  ...
```
Basically removing the versions and deduping the os names.
Then:
```
name, version = os_name.split("/")
if version != "-":
  return " ".join([capitalized_oses.get(name, name), version])
else:
  return capitalized_oses.get(name, name)
```

This way if there is a match, we get "Ubuntu 19.10" if there isn't a match we get "my-super-cool-distro 16" rather than "my-super-cool-distro/16"